src/components/Card.js

```diff
@@ -1,24 +1,47 @@
 export default class Card {
-  constructor(cardData, cardSelector, handleImageClick) {
+  constructor(
+    {
+      cardData,
+      handleImageClick,
+      handleDeleteButton,
+      handleCardLike,
+      handleCardUnlike,
+    },
+    cardSelector
+  ) {
     this._cardSelector = cardSelector;
     this._name = cardData.name;
     this._link = cardData.link;
+    this._id = cardData._id;
 
     this._handleImageClick = handleImageClick;
+    this._handleDeleteButton = handleDeleteButton;
+    this._handleCardLike = handleCardLike;
+    this._handleCardUnlike = handleCardUnlike;
   }
 
   _getTemplate() {
     return this._cardSelector.cloneNode(true);
   }
 
-  _handleLikeButton = () => {
-    const likeButton = this._likeButton;
-    likeButton.classList.toggle("card__like-button_active");
-  };
+  getId() {
+    return this._id;
+  }
 
-  _handleDeleteButton = () => {
+  removeItem() {
     this._element.remove();
-    this._element = null;
+  }
+
```

Comment:
Needs correction: you should update the likes button only after the successful api response

The code for rendering likes should look like this:

```js
// in Card class 

// call this method from index.js after successful api response in `then` block
setIsLiked(isLiked) { 
   // set instance variable
    this._isLiked = isLiked; 
    this._renderLikes(); 
} 

isLiked() {
  return this._isLiked;
}

_renderLikes() {
  if (this._isLiked) {
    // add active class to the like button
  } else {
    // remove active class from the like button
  }
}

getView() { 
  ...
  // render initial likes, you saved them in the constructor
  this._renderLikes()
}

// index.js
if (card.isLiked()) {
  api.dislikeCard(...).then(response => card.setIsLiked(response._isLiked)).catch(() => /*handle error*/)
} else {
 // the same for adding like
}
```